cmd/proxy_get.go

```diff
@@ -0,0 +1,148 @@
+list of supported GET queries.
+`
+
+const getCmdExample = `
+# Get the proxy config dump for the given pod 'bookbuyer-5ccf77f46d-rc5mg' in the 'bookbuyer' namespace
+osm proxy get config_dump bookbuyer-5ccf77f46d-rc5mg -n bookbuyer
+
+# Get the cluster config for the given pod 'bookbuyer-5ccf77f46d-rc5mg' in the 'bookbuyer' namespace and output to file 'clusters.txt'
+osm proxy get clusters bookbuyer-5ccf77f46d-rc5mg -n bookbuyer -f clusters.txt
+`
+
+type proxyGetCmd struct {
+	out        io.Writer
+	config     *rest.Config
+	clientSet  kubernetes.Interface
+	query      string
+	namespace  string
+	pod        string
+	localPort  uint16
+	outFile    string
+	sigintChan chan os.Signal
+}
+
+func newProxyGetCmd(config *action.Configuration, out io.Writer) *cobra.Command {
+	getCmd := &proxyGetCmd{
+		out:        out,
+		sigintChan: make(chan os.Signal, 1),
+	}
+
+	cmd := &cobra.Command{
+		Use:   "get QUERY POD",
+		Short: "get query for proxy",
+		Long:  getCmdDescription,
+		Args:  cobra.ExactArgs(2),
+		RunE: func(_ *cobra.Command, args []string) error {
+			getCmd.query = args[0]
+			getCmd.pod = args[1]
+			conf, err := config.RESTClientGetter.ToRESTConfig()
+			if err != nil {
+				return errors.Errorf("Error fetching kubeconfig: %s", err)
+			}
+			getCmd.config = conf
+
+			clientset, err := kubernetes.NewForConfig(conf)
+			if err != nil {
+				return errors.Errorf("Could not access Kubernetes cluster, check kubeconfig: %s", err)
+			}
+			getCmd.clientSet = clientset
+			return getCmd.run()
+		},
+		Example: getCmdExample,
+	}
+
+	//add mesh name flag
+	f := cmd.Flags()
+	f.StringVarP(&getCmd.namespace, "namespace", "n", metav1.NamespaceDefault, "Namespace of pod")
+	f.StringVarP(&getCmd.outFile, "file", "f", "", "File to write output to")
+	f.Uint16VarP(&getCmd.localPort, "local-port", "p", constants.EnvoyAdminPort, "Local port to use for port forwarding")
+
+	return cmd
+}
+
+func (cmd *proxyGetCmd) run() error {
+	// Check if the pod belongs to a mesh
+	pod, err := cmd.clientSet.CoreV1().Pods(cmd.namespace).Get(context.TODO(), cmd.pod, metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("Could not find pod %s in namespace %s", cmd.pod, cmd.namespace)
+	}
+	if !isMeshedPod(*pod) {
+		return fmt.Errorf("Pod %s in namespace %s is not a part of a mesh", cmd.pod, cmd.namespace)
+	}
+	if pod.Status.Phase != corev1.PodRunning {
+		return fmt.Errorf("Pod %s in namespace %s is not running", cmd.pod, cmd.namespace)
+	}
+
+	dialer, err := k8s.DialerToPod(cmd.config, cmd.clientSet, cmd.pod, cmd.namespace)
+	if err != nil {
+		return err
+	}
+
+	portForwarder, err := k8s.NewPortForwarder(dialer, fmt.Sprintf("%d:%d", cmd.localPort, constants.EnvoyAdminPort))
+	if err != nil {
+		return errors.Errorf("Error setting up port forwarding: %s", err)
+	}
+
+	err = portForwarder.Start(func(pf *k8s.PortForwarder) error {
+		defer pf.Stop()
+		url := fmt.Sprintf("http://localhost:%d/%s", cmd.localPort, cmd.query)
```

Comment:
I'd straight up hard-code `cmd.query` to be `config_dump?include_eds`
So this could be `url := fmt.Sprintf("localhost:%d/config_dump?include_eds", cmd.localPort)`